docs/EXPERIMENTAL_PUBLIC_TRANSPORT_SUPPORT.md

```diff
@@ -0,0 +1,79 @@
+
+
+## Instructions for turning on full public transport support
+
+1. Run the python script for collecting GTFS feeds from two GTFS aggregators: [Transitland](https://www.transit.land/) and [OpenMobilityData.](http://transitfeeds.com/feeds) You can find the script in omim/tools/python/transit/gtfs/. It is called download_gtfs.py. It crawls all the feeds from these aggregators, loads feed zips and extracts as subdirectories to the specified directory.
+
+Example:
+```
+python3 download_gtfs.py --path=dir_for_storing_feeds --mode=fullrun --source=all --omd_api_key=YOUR_KEY_FOR_OPEN_MOBILITY_DATA_API
+```
+
+In this example all accessible feeds from Transitland and OpenMobilityData will be downloaded to the `dir_for_storing_feeds` directory. But if you don't want to create key for OpenMobilityData API you can run this tool for crawling Transitland only:
+
+```
+python3 download_gtfs.py --path=dir_for_storing_feeds --mode=fullrun --source=transitland
+```
+
+After completing this step you will have directory containing subdirectories with GTFS feeds.
+
+2. Build and run the C++ tool for filtering, combining and transforming GTFS feeds (downloaded on the previous step) to the intermediate json files. Later they will be passed to the generator_tool for creating transit section in new format. You can find the tool in `omim/transit/world_feed/gtfs_converter.` The tool is` gtfs_converter.`
+
+It can be run in two modes: for handling GTFS-only or for merging data from GTFS and OSM.
+
+
+Example of running `gtfs_converter` for merging GTFS and OSM data:
+
+```
+./gtfs_converter --path_mapping=mapping.txt --path_mapping_edges=mapping_edges.txt --path_gtfs_feeds=dir_for_storing_feeds --path_subway_json=subway_latest.json --path_json=result_json_dir --path_resources=omim/data
+```
+
+Example of running `gtfs_converter` for saving only GTFS data (without OSM):
+
+```
+./gtfs_converter --path_mapping=mapping.txt --path_mapping_edges=mapping_edges.txt --path_gtfs_feeds=dir_for_storing_feeds --path_json=result_json_dir --path_resources=omim/data
+```
+
+:exclamation: GTFS data often changes: some routes are cancelled, some stops are built, timetables constantly change. **If you're planning to build maps periodically, more than once, then you must take care of the safety of files with transit entities ids which are generated by `gtfs_converter.`** The goal is to make transit entities consistent between re-runs of the tool so that routing between old regions and the new ones doesn't fail. It is better to backup files specified as `path_mapping` and `path_mapping_edges`.
+
+
+After completing this step you will have directory with line-by-line jsons in directory specified by `path_json` option of `gtfs_converter.`
+
+
+3. In map_generator.ini uncomment `SUBWAY_URL` parameter and set `TRANSIT_URL` parameter to the directory created on the previous step. Example:
```

Comment:
Do we really need both SUBWAY_URL and TRANSIT_URL parameters now? Doesn't gtfs_converter mix osm subway data somewhere into result_json_dir at the step 2?

Reply:
this is done on purpose to minimize risks of building production maps with wrong or absent section with transit data. In summer we had the plan: add new "experimental" code with minimal risks and minimal changes in existing code. This approach (SUBWAY_URL + TRANSIT_URL) looks stupid but it is the most safe one to make experiments with the new section.